addons/html_editor/static/src/main/link/link_popover.js

```diff
@@ -197,75 +200,45 @@ export class LinkPopover extends Component {
             )}`;
             this.state.previewIcon = true;
 
-            let metadata = {};
-            // Fetch the metadata
+            // Fetch the metadata and cache it
             try {
-                metadata = await rpc("/html_editor/link_preview_external", {
+                externalLinkCache[url] ||= await rpc("/html_editor/link_preview_external", {
                     preview_url: url,
                 });
             } catch {
                 // when it's not possible to fetch the metadata we don't want to block the ui
                 return;
             }
+            const externalMetadata = externalLinkCache[url];
 
-            this.state.urlTitle = metadata?.og_title || this.state.url;
-            this.state.urlDescription = metadata?.og_description || "";
-            this.state.imgSrc = metadata?.og_image || "";
-            if (metadata?.og_image && this.state.label && this.state.urlTitle === this.state.url) {
+            this.state.urlTitle = externalMetadata?.og_title || this.state.url;
+            this.state.urlDescription = externalMetadata?.og_description || "";
+            this.state.imgSrc = externalMetadata?.og_image || "";
+            if (
+                externalMetadata?.og_image &&
+                this.state.label &&
+                this.state.urlTitle === this.state.url
+            ) {
                 this.state.urlTitle = this.state.label;
             }
         } else {
-            await this.keepLastPromise
+            const html_parser = new window.DOMParser();
+            // Get
```

Comment:
I think you forgot to delete this comment line ("Get")

Reply:
Nope, I actually put it there to align with  the `//set` below :rofl:

Reply:
What do you mean by "GET" ? It s not really clear ?

Reply:
Just updated the comments, I used `set` and `get` as "titles" but can be a bit redundant :grimacing: